src/mono/wasm/runtime/roots.ts

```diff
@@ -381,7 +388,13 @@ class WasmExternalRoot<T extends MonoObject> implements WasmRoot<T> {
     }
 
     set(value: T): T {
-        cwraps.mono_wasm_write_managed_pointer_unsafe(this.__external_address, <ManagedPointer>value);
+        if (value === <any>0) {
+            // The expensive write barrier is only necessary when storing a managed pointer into
```

Comment:
I would prefer to make this change only directly into the `clear()` method.

Reply:
I'm fine with that, I'll change it to work that way